pkg/applier/status.go

```diff
@@ -119,10 +119,14 @@ func (m ObjectStatusMap) Log(logger infofLogger) {
 }
 
 func writeStatus(w io.Writer, status interface{ String() string }, ids []core.ID) {
+	var err error
 	if len(ids) == 0 {
-		fmt.Fprintf(w, "%s (%d)", status, len(ids))
```

Comment:
Should this be using the same logger as the rest of the file?

Reply:
huh? It's writing to the string builder passed into it. the builder gets logged and reset multiple times.

Reply:
My mistake, I misread this as logging

Reply:
I think this should probably return an error instead of panic in that case

Reply:
Writing to a string builder never fails. If it does, it's a programmer error and we'll want to catch it in testing.

```
func (b *Builder) Write(p []byte) (int, error) {
	b.copyCheck()
	b.buf = append(b.buf, p...)
	return len(p), nil
}
```

Reply:
That seems like an implementation detail that's not guaranteed by this function. This function expects an `io.Writer` interface, so it would be an easy mistake to make in the future if someone passed a different struct. We could change the argument to `*strings.Builder` to make that more clear

Reply:
done